3-UsingAPIs/Week2/homework/ex1-programmerFun/index.js

```diff
@@ -17,29 +17,37 @@ Full description at: https://github.com/HackYourFuture/Homework/blob/main/3-Usin
    url with `.shx`. There is no server at the modified url, therefore this 
    should result in a network (DNS) error.
 ------------------------------------------------------------------------------*/
-function requestData(url) {
-  // TODO return a promise using `fetch()`
+async function requestData(url) {
+  const response = await fetch(url);
+  if (!response.ok) {
+    throw new Error('HTTP error');
+  } else {
+    const data = await response.json();
+    return data;
```

Comment:
You do not need the `await` here since there is no  `try/catch` block here. You can immediately return the promise from `response.json`.

```suggestion
    return response.json();
```